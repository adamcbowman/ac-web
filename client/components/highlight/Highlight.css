@import '../../styles/colors.css';

.Highlight {
    position: fixed 85px 50px auto 50px;
    display: flex;
    text-align: center;
    margin: 0 auto;
    z-index: 2;
    color: white;
    font-weight: 700;
    background-color: var(--brand-secondary);

    & > *:first-child {
        flex: 1 0;
    }

    & a {
        color: white;
    }

    & button {
        color: white;
    }

    & .Close {
        padding: 0 1em;
    }
}

.Danger {
    composes: Highlight;
}

.Warning {
    composes: Highlight;

    background-color: var(--warning);
}

.Info {
    composes: Highlight;

    background-color: var(--info);
}

.Success {
    composes: Highlight;

    background-color: var(--success);
}
