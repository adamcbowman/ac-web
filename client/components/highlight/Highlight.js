import React from 'react'
import PropTypes from 'prop-types'
import CSSModules from 'react-css-modules'
import { Close } from 'components/button'
import styles from './Highlight.css'
import noop from 'lodash/noop'

export const DANGER = 'danger'
export const WARNING = 'warning'
export const INFO = 'info'
export const SUCCESS = 'success'

const STYLE_NAMES = new Map([
    [DANGER, 'Danger'],
    [WARNING, 'Warning'],
    [INFO, 'Info'],
    [SUCCESS, 'Success'],
])

Highlight.propTypes = {
    children: PropTypes.node.isRequired,
    style: PropTypes.oneOf([DANGER, WARNING, INFO, SUCCESS]).isRequired,
    dismissable: PropTypes.bool,
    onDismiss: PropTypes.func,
}

function Highlight({
    style = WARNING,
    dismissable,
    onDismiss = noop,
    children,
}) {
    return (
        <div styleName={STYLE_NAMES.get(style)}>
            {children}
            {dismissable && <Close transparent onClick={onDismiss} />}
        </div>
    )
}

export default CSSModules(Highlight, styles)
