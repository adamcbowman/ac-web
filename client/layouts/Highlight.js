import React, { PureComponent } from 'react'
import Component from 'components/highlight'
import { Banner } from 'components/application'
import { Link, StructuredText } from 'prismic/components/base'
import { Document } from 'prismic/new-containers'
import { highlight } from 'prismic/params'
import { SessionStorage } from 'services/storage'

// Constants
const STORAGE_KEY = 'highlight-hidden'

export default class Highlight extends PureComponent {
    storage = SessionStorage.create()
    state = {
        hidden: this.storage.get(STORAGE_KEY, false),
    }
    handleDismiss = () => {
        this.setState({ hidden: true }, () => {
            this.storage.set(STORAGE_KEY, true)
        })
    }
    children = ({ document }) => {
        if (!document || this.state.hidden) {
            return null
        }

        const { link, description, style } = document.data
        const content = <StructuredText value={description} />

        return (
            <Banner>
                <Component
                    style={style}
                    onDismiss={this.handleDismiss}
                    dismissable>
                    {link ? <Link {...link}>{content}</Link> : content}
                </Component>
            </Banner>
        )
    }
    render() {
        return <Document {...highlight()}>{this.children}</Document>
    }
}
