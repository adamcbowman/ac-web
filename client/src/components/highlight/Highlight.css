@import "../../styles/colors.css";

@custom-media --medium-screen-width (width > 600px);
@custom-media --large-screen-width (width > 800px);
@custom-media --massive-screen-width (width > 1000px);

.Highlight {
    position: fixed 85px 50px auto 50px;
    display: flex;
    text-align: center;
    margin: 0 auto;
    z-index: 1;
    border-style: solid;
    border-width: 1px;

    & > *:first-child {
        flex: 1 0;
        margin-left: 1em;
    }

    @media(--medium-screen-width) {
        right: 125px;
        left: 125px;
    }
    @media(--large-screen-width) {
        right: 200px;
        left: 200px;
    }
    @media(--massive-screen-width) {
        right: 250px;
        left: 250px;
    }
}

.Highlight--Danger {
    composes: Highlight;

    background-color: var(--danger);
    border-color: var(--danger-dark);
}

.Highlight--Warning {
    composes: Highlight;

    background-color: var(--warning);
    border-color: var(--warning-dark);
}

.Highlight--Info {
    composes: Highlight;

    background-color: var(--info);
    border-color: var(--info-dark);
}

.Highlight--Success {
    composes: Highlight;

    background-color: var(--success);
    border-color: var(--success-dark);
}
