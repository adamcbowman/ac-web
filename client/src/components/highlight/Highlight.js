import React, {PropTypes} from 'react'
import CSSModules from 'react-css-modules'
import capitalize from 'lodash/capitalize'
import {Close} from 'components/button'
import styles from './Highlight.css'

function noop() {}

Highlight.propTypes = {
    children: PropTypes.node.isRequired,
    style: PropTypes.oneOf(['danger', 'warning', 'info', 'success']).isRequired,
    dismissable: PropTypes.bool,
    onDismiss: PropTypes.func,
}

function Highlight({style = 'warning', dismissable, onDismiss = noop, children}) {
    return (
        <div styleName={`Highlight--${capitalize(style)}`}>
            {children}
            {dismissable && <Close onClick={onDismiss} />}
        </div>
    )
}

export default CSSModules(Highlight, styles)
