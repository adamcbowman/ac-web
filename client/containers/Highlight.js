import React, {Component} from 'react'
import PropTypes from 'prop-types'
import {connect} from 'react-redux'
import Highlight from '~/components/highlight'
import {InnerHTML} from '~/components/misc'
import {load} from '~/actions/prismic'
import format from 'date-fns/format'
import startOfTomorrow from 'date-fns/start_of_tomorrow'
import startOfYesterday from 'date-fns/start_of_yesterday'
import {Predicates} from '~/prismic'
import parser from '~/prismic/parser'
import {SessionStorage} from '~/services/storage'

@connect(null, {load})
export default class Container extends Component {
    static propTypes = {
        load: PropTypes.func.isRequired,
    }
    state = {
        highlight: null
    }
    constructor(props) {
        super(props)

        this.storage = SessionStorage.create()
    }
    set highlight(highlight = null) {
        this.setState({highlight})
    }
    get highlight() {
        return this.state.highlight
    }
    get hidden() {
        return this.storage.get('highlight-hidden-status')
    }
    set hidden(hidden) {
        return this.storage.set('highlight-hidden-status', hidden)
    }
    componentDidMount() {
        this.load()
    }
    load() {
        return this.props.load({
            type: 'highlight',
            predicates: [
                Predicates.dateBefore('my.highlight.start_date', format(startOfTomorrow(), 'YYYY-MM-DD')),
                Predicates.dateAfter('my.highlight.end_date', format(startOfYesterday(), 'YYYY-MM-DD')),
            ]
        }).then(response => {
            const {results: [highlight]} = response

            this.highlight = highlight ? parser.parse(highlight) : null
        })
    }
    handleDismiss = () => {
        this.hidden = true
        this.highlight = null
    }
    render() {
        if (!this.highlight || this.hidden === true) {
            return null
        }

        const {description, style} = this.highlight

        return (
            <Highlight style={style} onDismiss={this.handleDismiss} dismissable>
                <InnerHTML>
                    {description}
                </InnerHTML>
            </Highlight>
        )
    }
}
