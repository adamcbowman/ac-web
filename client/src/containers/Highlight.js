import React, {Component} from 'react'
import {compose, withHandlers, lifecycle, renameProp, withProps, withState} from 'recompose'
import {connect} from 'react-redux'
import moment from 'moment'
import Highlight from 'components/highlight'
import {InnerHTML} from 'components/misc'
import {load} from 'actions/prismic'
import {yesterday, tomorrow, formatAsDay} from 'utils/date'
import {Predicates} from 'prismic'
import parser from 'prismic/parser'
import {SessionStorage} from 'services/storage'

@connect(null, {load})
export default class Container extends Component {
    state = {
        highlight: null
    }
    constructor(props) {
        super(props)

        this.storage = SessionStorage.create()
    }
    set highlight(highlight = null) {
        this.setState({highlight})
    }
    get highlight() {
        return this.state.highlight
    }
    get hidden() {
        return this.storage.get('highlight-hidden-status')
    }
    set hidden(hidden) {
        return this.storage.set('highlight-hidden-status', hidden)
    }
    componentDidMount() {
        this.load()
    }
    load() {
        return this.props.load({
            type: 'highlight',
            predicates: [
                Predicates.dateBefore('my.highlight.start_date', formatAsDay(tomorrow())),
                Predicates.dateAfter('my.highlight.end_date', formatAsDay(yesterday())),
            ]
        }).then(response => {
            const {results: [highlight]} = response

            this.highlight = highlight ? parser.parse(highlight) : null
        })
    }
    handleDismiss = () => {
        this.hidden = true
        this.highlight = null
    }
    render() {
        if (!this.highlight || this.hidden === true) {
            return null
        }

        const {description, style, link} = this.highlight

        return (
            <Highlight style={style} onDismiss={this.handleDismiss} dismissable>
                <InnerHTML>
                    {description}
                </InnerHTML>
            </Highlight>
        )
    }
}
