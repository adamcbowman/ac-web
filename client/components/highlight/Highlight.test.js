import React from 'react'
import Renderer from 'react-test-renderer'
import Highlight, { DANGER, WARNING, INFO, SUCCESS } from './Highlight'

const TYPES = [DANGER, WARNING, INFO, SUCCESS]

test('highlight component', () => {
    TYPES.forEach(type => {
        const highlight = Renderer.create(
            <Highlight type={type}>Content</Highlight>
        )

        expect(highlight).toMatchSnapshot()
    })
})
