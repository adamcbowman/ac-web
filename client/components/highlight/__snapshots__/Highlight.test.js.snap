// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlight component 1`] = `
<div
  className="Danger"
  style={undefined}
>
  Content
</div>
`;

exports[`highlight component 2`] = `
<div
  className="Warning"
  style={undefined}
>
  Content
</div>
`;

exports[`highlight component 3`] = `
<div
  className="Info"
  style={undefined}
>
  Content
</div>
`;

exports[`highlight component 4`] = `
<div
  className="Success"
  style={undefined}
>
  Content
</div>
`;
